### PR TITLE
Update escrow to feature metadata, recipient can be set after creation

### PR DESCRIPTION
**Pull request by Callum-A**
After discussions with @JakeHartnell and @orkunkl we decided to expand functionality for escrow, this new functionality is as follows:

- Recipient is now optional upon creation, this allows Escrows to be used as bounties e.g. for bug bounties etc.
- Recipient can be set by the arbiter using the new SetRecipient message
- Escrows now feature additional meta data, including a title and description. E.g. for a bounty the description may include steps needed to be valid for said bounty.

Tests are added/updated accordingly.

Feedback as always is welcome

### CODE REVIEW

File: contracts/cw20-escrow/src/contract.rs [hunk 1]
```diff
@@ -116,6 +126,28 @@ pub fn execute_create(
...
+    if info.sender != escrow.arbiter {
+        return Err(ContractError::Unauthorized {});
+    }
+    let recipient = deps.api.addr_validate(recipient.as_str())?;
+    escrow.recipient = Some(recipient.clone());
+
+    ESCROWS.save(deps.storage, &id, &escrow)?;
+    Ok(Response::new().add_attributes(vec![
+        ("action", "set_recipient"),
+        ("id", id.as_str()),
```
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:146]:
```suggestion
        ("id", id),
```

File: contracts/cw20-escrow/src/contract.rs [hunk 2]
```diff
@@ -116,6 +126,28 @@ pub fn execute_create(
...
+        ("recipient", recipient.as_str()),
```
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:147]:
```suggestion
        ("recipient", recipient),
```

**Comment by Callum-A** [comment]
Changes all look good to me : )

**Comment by orkunkl** [comment]
Waiting for Circle CI to be configured by core team.

File: contracts/cw20-escrow/src/contract.rs [hunk 3]
```diff
@@ -428,6 +474,107 @@ mod tests {
         assert!(matches!(err, ContractError::Std(StdError::NotFound { .. })));
     }
 
+    #[test]
+    fn set_recipient_after_creation() {
+        let mut deps = mock_dependencies();
+
```
Comment (orkunkl) [inline contracts/cw20-escrow/src/contract.rs:480]:
```suggestion


```

**Comment by orkunkl** [comment]
Circle CI works now, can you fix the errors @Callum-A and merge?

**Review by orkunkl — APPROVED** [review]

**Comment by Callum-A** [comment]
Will do

**Comment by Callum-A** [comment]
@orkunkl any update on this? Build is fixed for escrow however other contracts are breaking it.

**Pull request merged**